java-checks/src/test/files/checks/SynchronizationOnGetClassCheck.java

```diff
@@ -0,0 +1,32 @@
+public class MyClass {
+  public void doSomethingSynchronized() {
+    synchronized (getClass()) { // Noncompliant [[sc=19;ec=29]] {{Synchronize on the static class name instead.}}
+      // ...
+    }
+  }
+
+  public void doSomethingSynchronized2() {
```

Comment:
Given the name of that method there's probably no need to put this second issue in a different method than the first one.